homeassistant/components/keba/sensor.py

```diff
@@ -0,0 +1,117 @@
+"""Support for KEBA charging station sensors."""
+import logging
+
+from homeassistant.const import ENERGY_KILO_WATT_HOUR
+from homeassistant.helpers.entity import Entity
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS = {
+    'Curr user': ['Max current', 'mdi:flash', 'A'],
+    'Setenergy': ['Energy target', 'mdi:gauge', ENERGY_KILO_WATT_HOUR],
+    'P': ['Charging power', 'mdi:flash', 'kW'],
+    'E pres': ['Session energy', 'mdi:gauge', ENERGY_KILO_WATT_HOUR],
+    'E total': ['Total Energy', 'mdi:gauge', ENERGY_KILO_WATT_HOUR],
+}
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station sensors")
+    sensors = []
+    keba = hass.data[DOMAIN]
+
+    for key, _ in sorted(SENSORS.items()):
+        sensors.append(KebaSensor(key, keba))
+
+    async_add_entities(sensors)
+
+
+class KebaSensor(Entity):
+    """The entity class for KEBA charging stations sensors."""
+
+    def __init__(self, key, keba):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._state = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Return False.
+
+        Data update is triggered from Keba entitiy.
+        """
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        name, _, _ = SENSORS.get(self._key, [None, None, None])
```

Comment:
See my above commentary about passing the values directly. Also, is there a chance that the key does not exist at all? And if so, should that cause erroring out instead of leaving the name as None?

Reply:
Changed according to your suggestion.